dotcom-rendering/src/web/components/SecureSignupIframe.importable.tsx

```diff
@@ -100,49 +101,54 @@ const postFormData = async (
 	});
 };
 
+type EventDescription =
+	| 'click-button'
+	| 'form-submission'
+	| 'submission-confirmed'
+	| 'submission-failed'
+	| 'open-captcha'
+	| 'captcha-load-error'
+	| 'form-submit-error'
+	| 'captcha-not-passed'
+	| 'captcha-passed';
+
 const sendTracking = (
 	newsletterId: string,
-	event:
-		| 'click-button'
-		| 'submit-form'
-		| 'form-submit-error'
-		| 'captcha-load-error'
-		| 'open-captcha'
-		| 'captcha-not-passed'
-		| 'captcha-passed'
-		| 'success-response-received'
-		| 'fail-response-received',
+	eventDescription: EventDescription,
 ): void => {
 	const ophanRecord = getOphanRecordFunction();
 
 	let action: OphanAction = 'CLICK';
-	const value = `${event} ${newsletterId}`;
 
-	switch (event) {
-		case 'submit-form':
-			action = 'SUBSCRIBE';
+	switch (eventDescription) {
+		case 'form-submission':
+		case 'captcha-not-passed':
+		case 'captcha-passed':
+			action = 'ANSWER';
 			break;
-		case 'open-captcha':
-			action = 'EXPAND';
+		case 'submission-confirmed':
+			action = 'SUBSCRIBE';
 			break;
-		case 'form-submit-error':
 		case 'captcha-load-error':
+		case 'form-submit-error':
+		case 'submission-failed':
 			action = 'CLOSE';
 			break;
-		case 'captcha-not-passed':
-		case 'captcha-passed':
-			action = 'ANSWER';
-			break;
-		case 'success-response-received':
-		case 'fail-response-received':
-			action = 'RETURN';
+		case 'open-captcha':
+			action = 'EXPAND';
 			break;
 		case 'click-button':
 		default:
 			action = 'CLICK';
 			break;
 	}
 
+	const value = JSON.stringify({
+		eventDescription,
+		newsletterId,
+		timestamp: formatTimestampToUTC(),
```

Comment:
The ophan/pageview data is pseudonymous so that shouldn't be an issue. I think they already get timestamps on when the pageview happened, so the level of precision isn't new.  The same level of detail is captured on the equivalent components in frontend.

The data team asked for timestamps so they can see how much time elapsed between each event. This is needed to understand the user behaviour - eg how long does it take people to sign up, did they have to complete a reCaptcha or not etc.

Reply:
> The feedback we've had from data privacy was that using ophan was preferable to the server-side routes available

Does this mean we've already checked with them and that they're happy to go ahead with capturing this extra data? If not, when adding new tracking, if there is a belief that the data could be considered personal, then we need to check with DP.

Reply:
My understanding is that event tracking within Ophan will only be registered if they are approved events, and will otherwise be dropped completely. Here the `OphanAction` type maps to a set of registered actions.

Can I ask whether there is a need for capturing the timestamp client-side, @dblatcher? We’ve found that readers’s clocks can be vastly out of sync, and it’s therefore preferable to rely on the timestamp on the receiving server.

Reply:
> My understanding is that event tracking within Ophan will only be registered if they are approved events, and will otherwise be dropped completely. Here the `OphanAction` type maps to a set of registered actions.
> 
This is true, but since the payload for the events includes a free text "value" field, they could be used to send extra data or context not anticipated when the approved events were designed, so there is still a need to consider privacy on how Ophan events are used, even with the constraints in place.

> Can I ask whether there is a need for capturing the timestamp client-side, @dblatcher? We’ve found that readers’s clocks can be vastly out of sync, and it’s therefore preferable to rely on the timestamp on the receiving server.
We're collecting the timestamp client-side because Ophan doesn't add timestamps to `ComponentEvents` server-side (it does add them to `PageView` events). Its not a specific need that they be captured client-side.

It might be feasible for Ophan to populate all `ComponentEvents` with a dedicated timestamp field server side - I expect it would bloat the total amount of data gathered, not sure if that would be a significant concern though. Happy to follow up on that option with the Ophan team?

The main use-case for the timestamps is more to assess the timing between events (e.g. "this user clicked the button, 30ms later the capchta script was executed, 5400ms after that, the user failed the capchta), so the user's clock being out of sync won't prevent this, but thanks for raising it - it could lead to some confusing data where it appears the pageView (server-side time) happened after the componentEvents (client-side time) within that pageView.

Reply:
> Does this mean we've already checked with them and that they're happy to go ahead with capturing this extra data? If not, when adding new tracking, if there is a belief that the data could be considered personal, then we need to check with DP.

Just to answer this point for the record - we did not consult DP on this PR, but are satisfied its within existing guidance.